Makefile

```diff
@@ -11,6 +11,8 @@ INSTALL = install
 INSTALL_DIR = $(INSTALL) -m755 -d
 INSTALL_DATA = $(INSTALL) -m644
 INSTALL_PROGRAM = $(INSTALL) -m755
+GIT_REVISION = $(shell (git rev-parse HEAD 2> /dev/null) || (cat .git/HEAD 2> /dev/null | head -c 40))
```

Comment:
On the other hand, I think it is fine to have git as a dependency for some make file commands, no? Otherwise, we will generate the wrong docs anyway.